main.go

```
@@ -5,93 +5,131 @@ import (
 		Ref:    inputs.targetRef,
 		Inputs: inputs.workflowInputs,
 	})
 
 	if err != nil {
 		githubactions.Fatalf("Error disptaching event: %v", err.Error())
 	}
+}
 
-	if inputs.waitForCheck {
-		// Wait for the check to be completed after all actions have passed
-		checkTimeoutCtx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(inputs.waitTimeoutSeconds))
-		// cancel context just in case it sticks around
-		defer cancel()
+func waitForCheck(client *github.Client, githubVars githubVars, inputs inputs, checkRun *github.CheckRun) {
+	if !inputs.waitForCheck {
+		githubactions.Infof("wait_for_check was false, proceeding\n")
+		return
+	}
+
+	checkTimeoutCtx, cancel := context.WithTimeout(context.Background(), time.Second*time.Duration(inputs.waitTimeoutSeconds))
+	defer cancel()
 
-		// wait for our own check to finish
-		err = checkWaiter(checkTimeoutCtx, client, owner, repo, int(*checkRun.ID))
-		if err != nil {
-			githubactions.Fatalf("Error waiting for created check to finish: %v", err.Error())
-		}
+	checkSucceeded, err := pollForCheckCompletion(checkTimeoutCtx, client, githubVars, inputs, int(*checkRun.ID))
+	if err != nil {
+		githubactions.Fatalf("Error waiting for check to finish: %v", err.Error())
+	}
 
-		githubactions.Infof("Check passed. Moving on.\n")
+	if checkSucceeded {
+		githubactions.Infof("Check completed successfully!\n")
 	} else {
-		githubactions.Infof("waitForCheck is false. Moving on.\n")
+		githubactions.Infof("Check failed!\n")
 	}
+
+}
+
+func scrapeOutputs(client *github.Client, githubVars githubVars, checkId int64) {
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	check, _, err := client.Checks.GetCheckRun(ctx, githubVars.repositoryOwner, githubVars.repositoryName, checkId)
+	if err != nil {
+		githubactions.Fatalf("Error fetching check for output scraping: %v", err.Error())
```

Since we will be scraping outputs multiple times, should we ignore any errors on individual scrapes?  It seems like this should only catch instances where the check run is accidentally deleted (since upon creation we get its ID)... and without the check run there is no way to signal the original job.  So it seems like we should either make even more noise in this case, or just ignore it until a few more attempts fail.

---

It doesn't scrape output multiple times and this is the original job. This is the original job scraping the output after it has already been signaled by distributor that the check is complete. As far as making more noise, this is about as much noise as you can make. This will fail the job in error with the given message (which appears in bright red and very clear in the logs).

---

OK.  I think I'd missed something with regard to fetching the check run status (which happens multiple times) vs getting the contents of the run itself.  For some reason I thought those were the same call